paper/src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/basic/SafeOreBreak.java

```diff
@@ -0,0 +1,199 @@
+import vg.civcraft.mc.civmodcore.inventory.items.MaterialUtils;
+import vg.civcraft.mc.civmodcore.players.settings.PlayerSetting;
+import vg.civcraft.mc.civmodcore.players.settings.PlayerSettingAPI;
+import vg.civcraft.mc.civmodcore.players.settings.gui.MenuSection;
+
+public final class SafeOreBreak extends BasicHack {
+
+	@AutoLoad
+	private List<String> ores;
+
+	private final List<BooleanOreSetting> oreSettings = new ArrayList<>();
+	private final List<List<Material>> breakOres = new ArrayList<>();
+
+	private final Set<Player> receivedWarning = Collections.newSetFromMap(new WeakHashMap<>());
+
+	public SafeOreBreak(final SimpleAdminHacks plugin, final BasicHackConfig config) {
+		super(plugin, config);
+	}
+
+	@Override
+	public void onEnable() {
+		super.onEnable();
+
+		MenuSection mainMenu = plugin.getSettingManager().getMainMenu();
+
+		OUTER:
+		for (String ore : ores) {
+			String[] parts = ore.split(",");
+			Material[] materialParts = new Material[parts.length];
+
+			for (int i = 0; i < parts.length; i++) {
+				String part = parts[i];
+				materialParts[i] = MaterialUtils.getMaterial(part);
+				if (materialParts[i] == null) {
+					logger.warning("Invalid material '"  + parts[0] + "'. Skipping.");
+					continue OUTER;
+				}
+			}
+
+			breakOres.add(Arrays.asList(materialParts));
+			String materialName = PlainTextComponentSerializer.plainText() .serialize(Component.translatable(materialParts[0]));
+			BooleanOreSetting setting = new BooleanOreSetting(plugin, true,
+					"Safe " + materialName + " break",
+					"safeOreBreak_" + materialParts[0].getKey().getKey(),
+					"Prevents you from breaking " + materialName
+							+ " without a silk touch pickaxe.",
+					materialParts[0]);
+			oreSettings.add(setting);
+			PlayerSettingAPI.registerSetting(setting, mainMenu);
+		}
+	}
+
+	@Override
+	public void onDisable() {
+		super.onDisable();
+	}
+
+	@EventHandler(ignoreCancelled = true)
+	public void onOreBreak(BlockBreakEvent event) {
+		if (event.getPlayer().getGameMode() != GameMode.SURVIVAL) {
+			return;
+		}
+
+		ItemStack item = event.getPlayer().getInventory().getItemInMainHand();
+		if (item.hasItemMeta() && item.getItemMeta().hasEnchant(Enchantment.SILK_TOUCH)) {
+			return;
+		}
+
+		OUTER:
+		for (int i = 0; i < breakOres.size(); i++) {
+			List<Material> list = breakOres.get(i);
+			for (Material material : list) {
+				if (material == event.getBlock().getType()) {
+					Boolean value = oreSettings.get(i).getValue(event.getPlayer());
+
+					if (!value) {
+						continue OUTER;
+					}
+
+					if (receivedWarning.add(event.getPlayer())) {
+						event.getPlayer().sendMessage(Component.text(
+										"A SimpleAdminHacks /config option is preventing you from breaking that ore without a silk touch pickaxe.")
+								.color(NamedTextColor.RED));
+					}
+					event.setCancelled(true);
+					return;
+				}
+			}
+		}
+	}
+
+	private static class BooleanOreSetting extends PlayerSetting<Boolean> {
```

Comment:
Just in case you're not aware, there's a [BooleanSetting](https://github.com/CivMC/CivModCore/blob/321ab959cfa72546551eb2600db6fa56e6bc742f/paper/src/main/java/vg/civcraft/mc/civmodcore/players/settings/impl/BooleanSetting.java) that you can use so much of the boilerplayer implemented below can be removed.

Reply:
That's what I used initially, but it doesn't allow you to change the icon so this is what this class is for.

Reply:
Ah, yeah, many of the player settings have pretty presumptuous constructors. Looking at the BooleanSetting, I can see that it doesn't allow you to create hidden boolean settings... weird.

Anyway, you can use [`#getVisualization()`](https://github.com/CivMC/CivModCore/blob/321ab959cfa72546551eb2600db6fa56e6bc742f/paper/src/main/java/vg/civcraft/mc/civmodcore/players/settings/PlayerSetting.java#L157-L162) which returns the icon-item as is, so you can modify it pretty much anyway you want. Just a suggestion.

Reply:
That won't work because the item needs to change based on the player and if the setting is enabled or not.

Reply:
Right.. you're doing all your item stuff in a fully custom and overriding `#getGuiRepresentation()`, which you can do by extending `BooleanSetting` too.

Reply:
I have changed it to extend BooleanSetting, thanks